client/src/main/java/io/split/storages/pluggable/adapters/UserCustomSplitAdapterConsumer.java

```diff
@@ -94,8 +96,20 @@ public List<String> splitNames() {
 
     @Override
     public Map<String, HashSet<String>> getNamesByFlagSets(List<String> flagSets) {
-        //Todo implement
-        return null;
+        Map<String, HashSet<String>> toReturn = new HashMap<>();
+        try {
+            UserPipelineWrapper pipelineExecution = _userStorageWrapper.pipeline();
+            for (String set: flagSets) {
+                 pipelineExecution.getMembers(PrefixAdapter.buildFlagSetPrefix(set));
+            }
+            List<Result> results = pipelineExecution.exec();
```

Comment:
Not sure if `results` can ever be `null`; in that case maybe we could initialize `results` as empty list.

Reply:
Updated. I added a check if it's null or empty.